app/models/importers/organisation.rb

```diff
@@ -11,19 +12,36 @@ def initialize(slug, batch: 10, start: 0)
   def run
     organisation = ::Organisation.create!(slug: slug)
     loop do
-      response = HTTParty.get "https://www.gov.uk/api/search.json?filter_organisations=#{slug}&count=#{batch}&fields=content_id&start=#{start}"
-      results = JSON.parse(response.body).fetch('results')
-
-      raise "No result for slug" if results.empty? && organisation.content_items.empty?
-
-      results.each do |result|
-        attributes = result.slice('content_id')
+      result = search_content_items_for_organisation
+      result.each do |content_item_attributes|
+        attributes = content_item_attributes.slice('content_id')
         organisation.content_items << ContentItem.new(attributes)
       end
 
-      break if results.length < batch
-      start += batch
+      break if last_page?(result)
+
+      next_page!
     end
+    raise 'No result for slug' if organisation.content_items.empty?
+  end
+
+  private
+
+  def last_page?(results)
+    results.length < batch
+  end
+
+  def next_page!
+    self.start += batch
```

Comment:
Doesn't `self` mean that you are using the class variable? Isn't `start` an instance variable now? Shouldn't `start` be passed to `next_page!` instead? Or, use `@start`?